requirements-extra.txt

```diff
@@ -1,3 +1,4 @@
-librosa
+librosa # for audio feature extraction
 sentencepiece>=0.0.6
 docopt
+tensorflow # for tensorboard visualization
```

Comment:
Ugh, I don't want to install all of tensorflow just to use tensorboard. I know this is the "extra" file, but people will need to install this for things like sentencepiece, which a large portion of the people using xnmt will need to install. Maybe we can add a brief document explaining how to do tensorboard visualization, which says at the beginning that you need to install tensorflow (if it's indeed necessary).

Reply:
Okay, I've reorganized these into a requirements-extra *directory* which contains separate files for the different extra use cases that can be installed via ```pip install -r ..```.